notebooks/machine_learning/raw/tut8.ipynb

```diff
@@ -0,0 +1,479 @@
+      "text/plain": [
+       "Index(['battery_power', 'blue', 'clock_speed', 'dual_sim', 'fc', 'four_g',\n",
+       "       'int_memory', 'm_dep', 'mobile_wt', 'n_cores', 'pc', 'px_height',\n",
+       "       'px_width', 'ram', 'sc_h', 'sc_w', 'talk_time', 'three_g',\n",
+       "       'touch_screen', 'wifi', 'price_range'],\n",
+       "      dtype='object')"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "data.columns"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We create our feature and targets the same as before using `train_test_split`. Creating and fitting the model is the same as well, except we're using `RandomForestClassifier` instead of `RandomForestRegressor`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set variables for the targets and features\n",
+    "y = data['price_range']\n",
+    "X = data.drop('price_range', axis=1)\n",
+    "\n",
+    "# Split the data into training and validation sets\n",
+    "train_X, val_X, train_y, val_y = train_test_split(X, y, random_state=7)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "RandomForestClassifier(bootstrap=True, class_weight=None, criterion='gini',\n",
+       "            max_depth=None, max_features='auto', max_leaf_nodes=None,\n",
+       "            min_impurity_decrease=0.0, min_impurity_split=None,\n",
+       "            min_samples_leaf=1, min_samples_split=2,\n",
+       "            min_weight_fraction_leaf=0.0, n_estimators=100, n_jobs=None,\n",
+       "            oob_score=False, random_state=7, verbose=0, warm_start=False)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Create the classifier and fit it to our training data\n",
+    "model = RandomForestClassifier(random_state=7, n_estimators=100)\n",
+    "model.fit(train_X, train_y)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The simplest metric for classification models is the **accuracy**, the number of correct predictions out of all predictions made. Scikit-learn provides `metrics.accuracy_score` to calculate this."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Accuracy: 0.864\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Predict classes given the validation features\n",
+    "pred_y = model.predict(val_X)\n",
+    "\n",
+    "# Calculate the accuracy as our performance metric\n",
+    "accuracy = metrics.accuracy_score(val_y, pred_y)\n",
+    "print(f\"Accuracy: {accuracy:.3f}\")"
```

Comment:
Most users have very little Python experience at this point, so they don't know f-strings.

We try to avoid showing things they don't understand or that make them worry they can't replicate what they've seen. Let's just print out full accuracy (despite the annoyingness of the extra sig-figures... so they don't feel like we're doing magic they can't replicate.